telegram/telegram.go

```diff
@@ -0,0 +1,740 @@
+		t.Bot.Handle(&btnShareBookmarks, t.handleShareBookmarks)
+		t.Bot.Handle(&btnGetOutputFile, t.handleGetOutput)
+		t.Bot.Handle(&btnDeleteHistory, t.handleDeleteHistory)
+
+		err = c.Send("به صفحه اصلی بازگشتید", userMenu)
+		return
+	})
+
+	err = c.Send("لطفا یک فیلتر را انتخاب کنید", filterMenu)
+	return
+}
+
+// TODO
+func (t *Telegram) handleShareBookmarks(c telebot.Context) error {
+	return nil
+}
+
+// TODO
+func (t *Telegram) handleGetOutput(c telebot.Context) error {
+	session := models.GetUserSession(c.Chat().ID)
+	getOutputFileMenu.Reply(
+		getOutputFileMenu.Row(btnGetAsZip, btnGetViaEmail),
+		getOutputFileMenu.Row(btnBackFilterMenu),
+	)
+
+	// TODO
+	t.Bot.Handle(&btnGetOutputFile, func(c telebot.Context) (err error) {
+		return
+	})
+
+	t.Bot.Handle(&btnGetViaEmail, func(c telebot.Context) (err error) {
+		session.State = "setting_user_email"
+		err = c.Send("لطفا ایمیل خود را وارد کنید")
+		return
+	})
+
+	t.Bot.Handle(&btnBackGetOutputFileMenu, func(c telebot.Context) (err error) {
+		session.State = ""
+		t.Bot.Handle(&btnSetFilters, t.handleSetFilters)
+		t.Bot.Handle(&btnShareBookmarks, t.handleShareBookmarks)
+		t.Bot.Handle(&btnGetOutputFile, t.handleGetOutput)
+		t.Bot.Handle(&btnDeleteHistory, t.handleDeleteHistory)
+
+		err = c.Send("به صفحه اصلی بازگشتید", userMenu)
+		return
+	})
+	return c.Send("نحوه خروجی گرفتن را انتخاب کنید", getOutputFileMenu)
+}
+
+func (t *Telegram) handleDeleteHistory(c telebot.Context) (err error) {
+	telegram_ID := c.Sender().ID
+	userRepo := repository.NewGormUser(database.GetInstnace().Db)
+	userAdsRepo := repository.NewGormUser_Ad(database.GetInstnace().Db)
+	user, e := userRepo.GetByTelegramId(strconv.Itoa(int(telegram_ID)))
+	if e != nil {
+		err = e
+		return
+	}
+	userAds, e := userAdsRepo.GetByUserId([]uint{user.ID})
+	if e != nil {
+		err = e
+		return
+	}
+	for _, item := range userAds {
+		e = userAdsRepo.Delete(item.User.ID, item.AdId)
+		if e != nil {
+			err = e
+			return
+		}
+	}
+
+	err = c.Send("تاریخچه شما حذف شد", userMenu)
+	t.Loggers.InfoLogger.Println("user history deleted")
+	return
+}
+
+// TODO Hirad: handle bookmark ad
+func (t *Telegram) handleBookmarkAd(c telebot.Context) (err error) {
+	return
+}
+
+// -adminMenu handlers
+// TODO
+func (t *Telegram) handleSeeCrawlDetails(c telebot.Context) (err error) {
+	return nil
+}
+
+// superAdmin menu handlers
+func (t *Telegram) handleAddAdmin(c telebot.Context) (err error) {
+	session := models.GetUserSession(c.Chat().ID)
+	session.State = "adding_admin"
+	err = c.Send("لطفا آیدی تلگرام کاربر مورد نظر خود را وارد کنید")
+	return
+}
+
+// TODO
+func (t *Telegram) handleManageAdmins(c telebot.Context) (err error) {
+	return nil
+}
+
+// TODO
+func (t *Telegram) handleSetCrawlTimeLimit(c telebot.Context) (err error) {
+	return nil
+}
+
+// TODO
+func (t *Telegram) handleSetNumberOfAds(c telebot.Context) (err error) {
+	return nil
+}
+
+func (t *Telegram) handleText(c telebot.Context) (err error) {
```

Comment:
By writing this function, you scared the ghost of Uncle Bob:)))
please write a cleaner function, your function is too long, so it is better to break it into smaller functions.